### DISCUSSION TITLE
How to play a single track each day from a playlist in sequential order?

### DISCUSSION THREAD

**Discussion by emtoonst** [AzuraCast/AzuraCast#7073] [Q&A] [answered]
I have a playlist with tracks, for each day of the year a specific track. How can I integrate this in Azuracast? My own ideas:

- Edit the Liquidsoap config to add functionality to play at a certain time the track of the day. By writing and reading the index to a file, I make it persistent.
- Have one file daily.mp3, and create a cronjob that replaces this file with the specific one. 

When I create a playlist with daily.mp3 and overwrite this file, it is also removed from the playlist. How can I fix this? Can I use a playlist from a remote URL that points to the daily.mp3 file? Like /var/azuracast/stations/my_radio_channel/media/Test/daily.mp3

Hope others have a suggestion as well.

**Comment by gAlleb** [comment, marked as answer]
Hi!
1. Create a folder that will hold the generated daily.mp3 file. e.g.  ```mkdir /var/azuracast/daily```
2. Create a cronjob as you like that replaces daily.mp3 file with the specific one in the above mentioned folder (a few hours before playback). Probably it should call some kind of bash script which eventually will do something like `cp daily.mp3 /var/azuracast/daily/` in the end.
3. Create or edit `docker-compose.override.yml` -> `nano /var/azuracast/docker-compose.override.yml` and mount a folder into azuracast:

```ruby
services:
  web:
    volumes:
      - /var/azuracast/daily:/var/azuracast/stations/***YOUR-STATION***/media/daily
```
4.  Restart azuracast.  
```
cd /var/azuracast
docker compose down
docker compose up -d
```
5.  Go to `Media -> Music Files`. You should see your `daily` folder there.
6. Create a new playlist (e.g. `DailyTracks`) and point it to your `daily` folder (this would help to prevent removal of the file from the playlist upon rewrite). Edit your playlist. Make it `Advanced`. Save changes.
7. Go to `Broadcasting -> Edit Liquidsoap Configuration -> Third Box` and paste:
```ruby
# We reload the playlist manually to be sure it holds the newest generated file. 
# It's nice to generate file at 9:00, reload playlist at 10:00 and play track a 11:00 for example to have some space between.

thread.when(predicate.activates({ 10h00m }), fun() -> playlist_dailytracks.reload())
```
Then do whatever you want. If it is an announcement of some sort and you wanna play it at 11:00 sharp  - you can `smooth_add` it on top of your usual playback:
```ruby
radio = smooth_add(duration=2., p=.3, normal=radio, special=switch([(predicate.once({ 11h00m }), playlist_dailytracks)]))
```
Alternatively you can use `switch` operator with or without `track_sensitive=false` (with or without interuption of the currently playing song):
```ruby
radio = fallback(track_sensitive=false,[
        switch([(predicate.once({11h00m}), playlist_dailytracks)]),
        radio
    ])
```

**Reply by emtoonst** [reply]
thanks gAlleb, I'll give it a try and let you know if this worked.

**Comment by emtoonst** [comment]
Hi @gAlleb, I have tested it and it works perfectly. Here's my bash script to copy the files, I run this as a daily cron job around midnight:

```bash
#!/bin/bash

# Get the day of the year (1-366)
DAY_OF_YEAR=$(date +%j)

# Define the source directory
SOURCE_DIR="/var/azuracast/YearFiles"

# Define the destination file
DEST_FILE="/var/azuracast/daily/daily.mp3"

# Find the file that matches the pattern
SOURCE_FILE=$(find "$SOURCE_DIR" -name "day-${DAY_OF_YEAR}-*.mp3" -print -quit)

# Check if the file exists
if [ -n "$SOURCE_FILE" ]; then
  # Copy the file to the destination
  sudo cp "$SOURCE_FILE" "$DEST_FILE"
else
  echo "File for day ${DAY_OF_YEAR} not found: dag-${DAY_OF_YEAR}-*.mp3" | sudo tee -a /var/azuracast/copy_daily_mp3.log
fi
```

**Reply by gAlleb** [reply]
Great!
